### PR TITLE
Improve app termination save pending

### LINKED ISSUE #971: Save game operation fails silently if user resigns and quits the game quickly enough
### Environment
- **FreeOrion Version**: 0.4.6 RC1
- **Operating System**: OSX
- **Fetched as**
  - Binary release
### Description

Something I never noticed before, probably because I very rarely play games that get so big that the save game process takes long enough:

When you initiate a save, the operation returns immediately, but as you can observe by the messages in the message window, the save game operation actually takes some time to complete. However, you are already able to resign and quit the game, in which case the save game operation gets silently aborted.

If you're like me and hide the message window, you don't realize that the save hasn't completed yet - you play your game, want to take a break and resume later, so you save, resign, quit. When you get back, your savegame isn't there, because you quit too quickly.
### Expected Result

If you save a game, one of the following should prevent quitting the game too quickly for the operation to finish:
- Simply don't return control to the user until the save operation has finished.
- Disable all buttons in the game menu that would quit the current game ("Resign", "Load") until the operation has finished, also block all other operations that could quit the application (closing the window etc.).
### Steps to reproduce

Huh, tedious:
- Play a big game long enough to get a save that takes long to finish.
- Initiate a save.
- Resign and quit as quickly as possible.
- The save should not have been created.

### PR DESCRIPTION
**Pull request by LGM-Doyle**
This improves/fixes [Issue #971 Save game operation fails silently if user resigns and quits the game quickly enough](https://github.com/freeorion/freeorion/issues/971).

The problem as described in the issue is that application can exit while the server is either collecting or writing the save game.

This PR solves this by popping up a dialog when there is a pending save and the user tries to return to the intro menu or completely exit the game.  The dialog says "A save game is in progress.  Click Stop Waiting to abandon the save."

If the player waits the application will quit/exit when the save completes.

If the player clicks "Stop Waiting" then the app will continue with the exit, without waiting for the save to complete.

A deficiency is that when the player elects to stop waiting the exit may wait  for 5 seconds and then kill the server.  Neither the server process or the AI processes are interruptible.  However, the player will have been informed and chosen to abandon the save and kill the server. 

I think this fixes the bug.

### CODE REVIEW

File: client/human/HumanClientApp.cpp [hunk 1]
```diff
@@ -1117,18 +1125,46 @@ void HumanClientApp::ResetClientData() {
...
+            GG::ThreeButtonDlg(
+                GG::X(320), GG::Y(200), UserString("SAVE_GAME_IN_PROGRESS"),
+                ClientUI::GetFont(ClientUI::Pts()+2),
+                ClientUI::WndColor(), ClientUI::WndOuterBorderColor(),
+                ClientUI::CtrlColor(), ClientUI::TextColor(), 1,
+                UserString("STOP_WAITING"))
+        { GG::Connect(save_completed_signal, &SaveGamePendingDialog::SaveCompletedHandler, this); }
+
+        void SaveCompletedHandler()
+        {
```
Comment (geoffthemedio) [inline client/human/HumanClientApp.cpp]: { to previous line

File: client/human/HumanClientApp.cpp [hunk 2]
```diff
@@ -1117,18 +1125,46 @@ void HumanClientApp::ResetClientData() {
...
+            DebugLogger() << "SaveGamePendingDialog:: save game completed handled.";
```
Comment (geoffthemedio) [inline client/human/HumanClientApp.cpp]: double ::  ?

File: default/stringtables/en.txt [hunk 3]
```diff
@@ -899,6 +899,12 @@ Error when reading save file.
 UNABLE_TO_SAVE_NOW_TRY_AGAIN
 Unable to save during AI processing. Try again when AIs have finished turns.
 
+SAVE_GAME_IN_PROGRESS
+A save game is in progress.  Click Stop Waiting to abandon the save.
+
+STOP_WAITING
+Stop waiting.
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]: Maybe something stronger-sounding? Like `Abort Save and Quit Immediately`

Comment (geoffthemedio) [reply default/stringtables/en.txt]: `Abort Save and Quit` is ambiguous. Will the quit be aborted?

Comment (Vezzra) [reply default/stringtables/en.txt]: So maybe "Abort Save, and Quit", or "Abort Save, then Quit"?

Comment (LGM-Doyle) [reply default/stringtables/en.txt]: I've split the messages in separate ones for return to main menu and exiting. 
`Return to Main Menu without saving.`
and
`Exit FreeOrion without saving.`

This is the same phrasing as the `RETURN_TO_INTRO` message.

**Pull request merged**